x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Limit.java

```diff
@@ -21,21 +21,37 @@ public class Limit extends UnaryPlan {
     public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(LogicalPlan.class, "Limit", Limit::new);
 
     private final Expression limit;
-
-    public Limit(Source source, Expression limit, LogicalPlan child) {
+    /**
+     * Important for optimizations. This should be {@code true} in most cases, which allows this instance to be duplicated past a child plan
+     * node that increases the number of rows, like for LOOKUP JOIN and MV_EXPAND.
+     * Needs to be set to {@code false} in {@link org.elasticsearch.xpack.esql.optimizer.rules.logical.PushDownAndCombineLimits} to avoid
+     * infinite loops from adding a duplicate of the limit past the child over and over again.
+     */
+    private final boolean allowDuplicatePastExpandingNode;
```

Comment:
How about renaming to something reusable such as `synthetic`, `duplicated`, `artificial`. This property is not inherent to Limit rather it's an artifact of planning that we're storing here for the time being.
We used it for mv_expand, now lookup join - it looks like there could be a pattern.

To signal that it's not serialized, make it `transient`.

Reply:
Thanks for the suggestion!

++ to marking as transient.

As for renaming: I'd like to refrain from very general terms like `synthetic` because that doesn't convey what it really means - but let's go with something that's less super duper verbose than the current name, but still precise. I like `duplicated`, so let's go with this.